Modules/Core/Common/src/itkNumberToString.cxx

```diff
@@ -16,45 +16,66 @@
  *
  *=========================================================================*/
 #include "itkNumberToString.h"
-#include "itkNumericTraits.h"
-#include "double-conversion/double-conversion.h"
+#include "double-conversion/double-to-string.h"
 
 #include <sstream>
 
-namespace itk
+namespace
 {
 
-template <>
+bool
+ConvertToShortest(const double_conversion::DoubleToStringConverter & converter,
+                  const double                                       val,
+                  double_conversion::StringBuilder &                 builder)
+{
+  return converter.ToShortest(val, &builder);
+}
+
+bool
+ConvertToShortest(const double_conversion::DoubleToStringConverter & converter,
+                  const float                                        val,
+                  double_conversion::StringBuilder &                 builder)
+{
+  // Call the converter member function that is specific for single-precision `float`.
+  return converter.ToShortestSingle(val, &builder);
+}
+
+template <typename TValue>
 std::string
-NumberToString<double>::operator()(double val) const
+FloatingPointNumberToString(const TValue val)
 {
-  char                                               buf[256];
-  const double_conversion::DoubleToStringConverter & converter =
-    double_conversion::DoubleToStringConverter::EcmaScriptConverter();
+  // Declare a buffer, large enough for strings like:
+  // "-100000000000000000000" (-1e20, either float or double, 23 chars)
+  // "-1.7976931348623157e+308" (-DBL_MAX, 25 chars)
+  // "-0.0000033333333333333333" (-3e-005/9.0, 26 chars)
+  char buf[32];
```

Comment:
Did you consult [this SO topic](https://stackoverflow.com/questions/1701055/what-is-the-maximum-length-in-chars-needed-to-represent-any-double-value)? Can you explain in the comments why not e.g. 325?

Reply:
You could add the other two cases from the comment and some of those edge cases from the SO answers as unit tests. That way we wouldn't have to guess how long of a buffer we need.

Reply:
> Did you consult [this SO topic](https://stackoverflow.com/questions/1701055/what-is-the-maximum-length-in-chars-needed-to-represent-any-double-value)? Can you explain in the comments why not e.g. 325?

@dzenanz  Nice SO topic, thanks!  `325` would make sense if the decimal (C printf "%f") notation is used for any floating point value. If I understand SO correctly. But Google's `EcmaScriptConverter().ToShortest(val)` switches to scientific (exponential) notation when the decimal notation gets too long.

As far as I can see (having looked at various edge cases), `EcmaScriptConverter().ToShortest(val)` never exceeds 26 chars. The original `buf[256]` is certainly much larger than necessary. So I think `buf[26]` should be OK, but I made it `buf[32]`, just to be sure. And because `32` is a power of two, just like `256` in the original code. 😃 

Note that `ToShortest(val)` would have an assert failure on some of those edge cases if the buffer would be smaller than 26:
> Assertion failed: !is_finalized() && position_ + n < buffer_.length(), file ITK\Modules\ThirdParty\DoubleConversion\src\double-conversion\utils.h, line 288

Before extending the unit tests or the comments, do you agree that `buf[32]` is fine?

Reply:
Can the plain notation (0.000000001) be requested? If yes, then we need a longer buffer. If not, 32 is a good choice.

Reply:
> Can the plain notation (0.000000001) be requested?

`itk::NumberToString` does not offer any way to change its floating point format. Internally it uses a `const` reference to the `static` DoubleToStringConverter object returned by `EcmaScriptConverter()`. The properties of this converter object cannot be changed. So the answer is _no_!  

> If yes, then we need a longer buffer. If not, 32 is a good choice.

Thanks 😃 

FYI, Below here the `NumberToString<double>{}(d)` output for `d` = `-1.0/9.0`, `-1.0/90.0`, `-1.0/900.0`, `-1.0/9000.0`, etc. Always below 26 chars:

    "-0.1111111111111111"
    "-0.011111111111111112"
    "-0.0011111111111111111"
    "-0.00011111111111111112"
    "-0.000011111111111111112"
    "-0.000001111111111111111"
    "-1.1111111111111111e-7"
    "-1.1111111111111112e-8"
    "-1.111111111111111e-9"
    "-1.1111111111111111e-10"
	...
    "-1.1111111111111113e-306"
    "-1.1111111111111112e-307"
    "-1.1111111111111113e-308"

Reply:
Thanks for answers.